exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -691,57 +746,198 @@ func (ku *Kucoin) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Subm
 	if err != nil {
 		return nil, err
 	}
+	var o string
 	switch s.AssetType {
 	case asset.Futures:
-		o, err := ku.PostFuturesOrder(ctx, &FuturesOrderParam{
+		if s.Leverage == 0 {
+			s.Leverage = 1
+		}
+		var stopOrderType, stopOrderBoundary string
+		switch s.Type {
+		case order.Stop, order.StopLimit, order.TrailingStop:
+			if s.TriggerPrice == 0 {
+				break
+			}
+			switch s.TriggerPriceType {
+			case order.IndexPrice:
+				stopOrderType = "IP"
+			case order.MarkPrice:
+				stopOrderType = "MP"
+			case order.LastPrice:
+				stopOrderType = "TP"
+			}
+			switch s.Type {
+			case order.StopLimit:
+				switch s.Side {
+				case order.Sell:
+					stopOrderBoundary = "up"
+				case order.Buy:
+					stopOrderBoundary = "down"
+				}
+			case order.TrailingStop, order.Stop:
+				switch s.Side {
+				case order.Sell:
+					// Stop-loss when order type is order.Stop
+					stopOrderBoundary = "down"
+				case order.Buy:
+					// Take Profit when order type is order.Stop
+					stopOrderBoundary = "up"
+				}
+			}
+		}
+		o, err = ku.PostFuturesOrder(ctx, &FuturesOrderParam{
 			ClientOrderID: s.ClientOrderID,
 			Side:          sideString,
 			Symbol:        s.Pair,
 			OrderType:     s.Type.Lower(),
 			Size:          s.Amount,
 			Price:         s.Price,
-			StopPrice:     s.TriggerPrice,
 			Leverage:      s.Leverage,
+			VisibleSize:   0,
 			ReduceOnly:    s.ReduceOnly,
 			PostOnly:      s.PostOnly,
-			Hidden:        s.Hidden})
+			Hidden:        s.Hidden,
+			Stop:          stopOrderBoundary,
+			StopPrice:     s.TriggerPrice,
+			StopPriceType: stopOrderType,
```

Comment:
Iceberg is also supported here too:
https://www.kucoin.com/docs/rest/futures-trading/orders/place-order